entry.go

```diff
@@ -33,7 +32,14 @@ type Entry struct {
 		Data     []byte
 		Identity []byte
 		LeafHash []byte
+		Index    *uint64
 	}
+
+	// indexFunc will be called when an index is assigned to the entry.
```

Comment:
I think this function MUST NOT modify the assigned index?

Reply:
Added a comment to that effect.